src/main/groovy/com/blackbaud/templates/tasks/AsyncProject.groovy

```diff
@@ -0,0 +1,192 @@
+    }
+
+    void addInternalTopic(String topicName, boolean sessionEnabled) {
+        addTopic(topicName, true, true, true, sessionEnabled)
+    }
+
+    void addExternalTopic(String topicName, boolean consumer, boolean publisher, boolean sessionEnabled) {
+        addTopic(topicName, false, publisher, consumer, sessionEnabled)
+    }
+
+    private void addTopic(String topicName, boolean internal, boolean consumer, boolean publisher, boolean sessionEnabled) {
+        initServiceBusIfNotAlreadyInitialized()
+
+        ServiceBusNameResolver formatter = new ServiceBusNameResolver(topicName)
+
+        File applicationPropertiesFile = basicProject.getProjectFile("src/main/resources/application-local.properties")
+        if (applicationPropertiesFile.text.contains("servicebus.namespace") == false) {
+            applicationPropertiesFile.append("""
+    servicebus.namespace=namespace
+""")
+        }
+        applicationPropertiesFile.append("""
+servicebus.${formatter.topicNameSnakeCase}.entity_path=${formatter.topicNameSnakeCase}
+servicebus.${formatter.topicNameSnakeCase}.shared_access_key_name=keyName
+servicebus.${formatter.topicNameSnakeCase}.shared_access_key=key
+""")
+        if (consumer) {
+            applicationPropertiesFile.append("""
+servicebus.${formatter.topicNameSnakeCase}.subscription=consumer
+""")
+        }
+
+        File serviceBusConfigFile = basicProject.findFile("ServiceBusConfig.java")
+        FileUtils.appendToClass(serviceBusConfigFile, """
+    @Bean
+    public ServiceBusProperties ${formatter.topicNameCamelCase}ServiceBusProperties(
+            @Value("\${servicebus.namespace}") String namespace,
+            @Value("\${servicebus.${formatter.topicNameSnakeCase}.entity_path}") String entityPath,
+            @Value("\${servicebus.${formatter.topicNameSnakeCase}.subscription}") String subscription,
+            @Value("\${servicebus.${formatter.topicNameSnakeCase}.shared_access_key_name}") String sasKeyName,
+            @Value("\${servicebus.${formatter.topicNameSnakeCase}.shared_access_key}") String sasKey) {
+        return ServiceBusProperties.builder()
+                .namespace(namespace)
+                .entityPath(entityPath)
+                .subscription(subscription)
+                .sharedAccessKey(sasKey)
+                .sharedAccessKeyName(sasKeyName)
+                .build();
+    }
+""")
+
+        File publisherConfigFile = publisher ? serviceBusConfigFile : basicProject.findComponentTestConfig()
+        FileUtils.appendToClass(publisherConfigFile, """
+    @Bean
+    public JsonMessagePublisher ${formatter.topicNameCamelCase}Publisher(
+            ServiceBusPublisherFactory serviceBusPublisherFactory,
+            @Qualifier("${formatter.topicNameCamelCase}ServiceBusProperties") ServiceBusProperties serviceBusProperties) {
+        return serviceBusPublisherFactory.createJsonPublisher(serviceBusProperties);
+    }
+""")
+
+        if (consumer) {
+            if (internal) {
```

Comment:
This is a bit confusing, having previously asserted the options were mutually exclusive. Maybe a comment on `addInternalTopic` just to highlight for an internal topic both publisher and consumer will be true as well?